data-serving/data-service/src/model/case.ts

```diff
@@ -1,43 +1,293 @@
 import mongoose from 'mongoose';
+import { ObjectId, Double } from 'mongodb';
 
-export const validOutcomes = ['pending', 'recovered', 'death'];
+export enum Sex {
+    Female,
+    Male,
+    Other,
+}
 
-const demographicsSchema = new mongoose.Schema({
-    age: {
+export enum Outcome {
+    Pending,
+    Recovered,
+    Death,
+}
+
+export enum Species {
+    HomoSapien,
+}
+
+export enum TravelPurpose {
+    Family,
+    Conference,
+    Work,
+}
+
+const dateValidator = {
+    type: Date,
+    min: '2019-11-01',
+    max: Date.now,
+};
+
+const ageRangeSchema = new mongoose.Schema({
+    minimum: {
+        type: Number,
+        min: -1,
+        max: 200,
+    },
+    maximum: {
         type: Number,
         min: 0,
         max: 200,
     },
+});
+
+const ageSchema = new mongoose.Schema({
+    years: {
+        type: Number,
+        min: -0.75,
+        max: 200,
+    },
+    range: ageRangeSchema,
+});
+
+const demographicsSchema = new mongoose.Schema({
+    ageSchema,
+    species: {
+        type: String,
+        enum: Object.keys(Species),
+    },
     sex: {
         type: String,
-        enum: ['female', 'male'],
+        enum: Object.keys(Sex),
     },
 });
 
-const caseSchema = new mongoose.Schema({
-    date: {
-        type: Date,
-        min: '2019-11-01',
-        max: Date.now,
-        required: 'Enter a date.',
+const eventSequenceSchema = new mongoose.Schema({
+    onsetSymptoms: dateValidator,
+    confirmed: {
+        ...dateValidator,
+        ...{
+            required: 'Enter a confirmation date',
```

Comment:
This still isn't required in the Case body -- is it? I think this makes the field required in eventSequenceSchema, but it's OK for eventSequenceSchema to be absent from the Case body.

Reply:
oops, good catch, fixed